src/core/hiwire.c

```diff
@@ -342,6 +342,11 @@ EM_JS(bool, hiwire_is_function, (JsRef idobj), {
   // clang-format on
 });
 
+EM_JS(bool, hiwire_function_supports_kwargs, (JsRef idfunc), {
+  let funcstr = Module.hiwire.get_value(idfunc).toString();
```

Comment:
You ought to handle the case where the function definition is not available, e.g. it is a wasm function. I think in this case you can assume it does not support kwargs.

Reply:
That case works automatically actually:
```javascript
>>> pyodide._module.asm.__PyAST_Optimize.toString()
"function 2738() { [native code] }"
```
Maybe I should add comment?

Reply:
That makes sense